app/src/androidTest/java/org/mozilla/fenix/ui/MediaNotificationTest.kt

```diff
@@ -0,0 +1,128 @@
+import org.mozilla.fenix.helpers.HomeActivityTestRule
+import org.mozilla.fenix.helpers.TestAssetHelper
+import org.mozilla.fenix.ui.robots.browserScreen
+import org.mozilla.fenix.ui.robots.homeScreen
+import org.mozilla.fenix.ui.robots.mDevice
+import org.mozilla.fenix.ui.robots.navigationToolbar
+
+/**
+ *  Tests for verifying basic functionality of media notifications:
+ *  - video and audio playback system notifications appear and can pause/play the media content
+ *  - a media notification icon is displayed on the homescreen for the tab playing media content
+ */
+class MediaNotificationTest {
+    /* ktlint-disable no-blank-line-before-rbrace */ // This imposes unreadable grouping.
+
+    private lateinit var mockWebServer: MockWebServer
+
+    @get:Rule
+    val activityTestRule = HomeActivityTestRule()
+
+    @Before
+    fun setUp() {
+        mockWebServer = MockWebServer().apply {
+            setDispatcher(AndroidAssetDispatcher())
+            start()
+        }
+    }
+
+    @After
+    fun tearDown() {
+        mockWebServer.shutdown()
+    }
+
+    @Test
+    fun videoPlaybackSystemNotificationTest() {
+        val videoTestPage = TestAssetHelper.getVideoPageAsset(mockWebServer)
+
+        navigationToolbar {
+        }.enterURLAndEnterToBrowser(videoTestPage.url) {
+            clickMediaPlayerPlayButton()
+            waitForPlaybackToStart()
+        }.openNotificationShade {
+            verifySystemNotificationExists(videoTestPage.title)
+            clickMediaSystemNotificationControlButton("Pause")
+            verifyMediaSystemNotificationButtonState("Play")
+        }
+
+        mDevice.pressBack()
+
+        browserScreen {
+            verifyMediaIsPaused()
+        }
+    }
+
+    @Test
+    fun audioPlaybackSystemNotificationTest() {
+        val audioTestPage = TestAssetHelper.getAudioPageAsset(mockWebServer)
+
+        navigationToolbar {
+        }.enterURLAndEnterToBrowser(audioTestPage.url) {
+            verifyPageContent(audioTestPage.content)
+            clickMediaPlayerPlayButton()
+            waitForPlaybackToStart()
+        }.openNotificationShade {
+            verifySystemNotificationExists(audioTestPage.title)
+            clickMediaSystemNotificationControlButton("Pause")
+            verifyMediaSystemNotificationButtonState("Play")
+        }
+
+        mDevice.pressBack()
+
+        browserScreen {
+            verifyMediaIsPaused()
+        }
+    }
+
+    @Test
+    fun tabMediaControlButton() {
+        val audioTestPage = TestAssetHelper.getAudioPageAsset(mockWebServer)
+
+        navigationToolbar {
+        }.enterURLAndEnterToBrowser(audioTestPage.url) {
+            verifyPageContent(audioTestPage.content)
+            clickMediaPlayerPlayButton()
+            waitForPlaybackToStart()
+        }.openHomeScreen {
+            verifyTabMediaControlButtonState("Pause")
+            clickTabMediaControlButton()
+            verifyTabMediaControlButtonState("Play")
```

Comment:
per our conversation, proceed to tab and verify if delay still happens w/ debugger. Try mDeviceWaitNotNul, but may need to add an idlingResource here.  @AaronMT can help

Reply:
added the step to verify the media state in the tab.